contracts/modules/cover/Cover.sol

```diff
@@ -234,179 +243,23 @@ contract Cover is ICover, MasterAwareV2 {
-
+    BuyCoverParams memory buyCoverParams,
+    CoverChunkRequest[] memory coverChunkRequests
+  ) external payable onlyMember returns (uint /*coverId*/) {
 
     CoverData memory cover = covers[coverId];
     require(cover.start + cover.period > block.timestamp, "Cover: cover expired");
 
-    uint premiumInAsset = _increasePeriod(coverId, periodExtension, paymentAsset);
-    require(premiumInAsset <= maxPremiumInAsset, "Cover: Price exceeds maxPremiumInAsset");
+    uint32 remainingPeriod = cover.start + cover.period - uint32(block.timestamp);
 
-    retrievePayment(premiumInAsset, covers[coverId].payoutAsset);
-  }
+    (, uint8 paymentAssetDecimals, ) = pool().assets(buyCoverParams.paymentAsset);
 
-  function _increasePeriod(uint coverId, uint32 periodExtension, uint8 paymentAsset) internal returns (uint) {
+    CoverChunk[] storage originalCoverChunks = coverChunksForCover[coverId];
 
-    CoverData storage cover = covers[coverId];
-    CoverChunk[] storage coverChunks = coverChunksForCover[coverId];
-
-    uint extraPremiumInNXM = 0;
-    for (uint i = 0; i < coverChunks.length; i++) {
-      IStakingPool stakingPool = IStakingPool(coverChunks[i].poolAddress);
-
-      (uint basePrice, uint premiumInNXM) = getPrice(
-        coverChunks[i].coverAmountInNXM, periodExtension,
-        cover.productId,
-        stakingPool
-      );
-
-      lastPrices[cover.productId][address(stakingPool)] = LastPrice(uint96(basePrice), uint32(block.timestamp));
-
-      stakingPool.extendPeriod(
-        cover.productId,
-        cover.period,
-        cover.start,
-        REWARD_BPS * coverChunks[i].premiumInNXM / BASIS_PRECISION,
-        periodExtension,
-        REWARD_BPS * (coverChunks[i].premiumInNXM + premiumInNXM) / BASIS_PRECISION,
-        cover.amount
-      );
-
-      extraPremiumInNXM += premiumInNXM;
-      coverChunks[i].premiumInNXM += uint96(premiumInNXM);
-    }
-
-    uint premiumInAsset = extraPremiumInNXM * pool().getTokenPrice(paymentAsset) / 1e18;
-
-    cover.period += periodExtension;
-
-    return premiumInAsset;
-  }
-
-  function increaseAmountAndReducePeriod(
-    IncreaseAmountAndReducePeriodParams memory params,
-    CoverChunkRequest[] calldata coverChunkRequests
-  ) external payable onlyMember returns (uint) {
-
-    CoverData storage cover = covers[params.coverId];
-    require(cover.start + cover.period > block.timestamp, "Cover: Cover expired");
-
-    require(
-      cover.period - (block.timestamp - cover.start) > params.periodReduction,
-      "Cover: periodReduction > remaining period"
-    );
-
-    // TODO: add check for max 20% MCR check for activeCoverAmountInNXM
-
-    CoverChunk[] storage originalCoverChunks = coverChunksForCover[params.coverId];
-
-    // reduce period
+    uint totalPreviousCoverAmountInNXM = 0;
+    // rollback previous cover
     for (uint i = 0; i < originalCoverChunks.length; i++) {
```

Comment:
Store originalCoverChunks.length in memory before the loop so yout don't read from storage at every iteration.